From 6b34e377a065309fae0f205f2e7b38173c134a78 Mon Sep 17 00:00:00 2001
From: Bernd Waibel <waebbl@gmail.com>
Date: Mon, 20 Jan 2020 15:53:23 +0100
Subject: [PATCH] CMakeLists.txt: enable HAVE_GL_* variables

Gentoo-specific hack: Enable HAVE_GL_* variables without
checking, because the checks fail for some yet unknown reason.
We assure that the headers are available, by RDEPEND on
virtual/opengl and virtual/glu.

Signed-off-by: Bernd Waibel <waebbl@gmail.com>
---
 CMakeLists.txt | 12 +++++++++---
 1 file changed, 9 insertions(+), 3 deletions(-)

diff --git a/CMakeLists.txt b/CMakeLists.txt
index f32f6ff..95d7e85 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -604,9 +604,15 @@ elseif(APPLE AND NOT COIN_BUILD_MAC_X11)
   check_include_file(OpenGL/glext.h HAVE_OPENGL_GLEXT_H)
   check_include_file(OpenGL/CGLCurrent.h HAVE_OPENGL_CGLCURRENT_H)
 else()
-  check_include_files("GL/gl.h" HAVE_GL_GL_H)
-  check_include_files("GL/glu.h" HAVE_GL_GLU_H)
-  check_include_files("GL/gl.h;GL/glext.h" HAVE_GL_GLEXT_H)
+  if(CMAKE_BUILD_TYPE MATCHES "Gentoo")
+    set(HAVE_GL_GL_H 1)
+    set(HAVE_GL_GLU_H 1)
+    set(HAVE_GL_GLEXT_H 1)
+  else()
+    check_include_files("GL/gl.h" HAVE_GL_GL_H)
+    check_include_files("GL/glu.h" HAVE_GL_GLU_H)
+    check_include_files("GL/gl.h;GL/glext.h" HAVE_GL_GLEXT_H)
+  endif()
 endif()
 
 set(CMAKE_REQUIRED_INCLUDES)
-- 
2.25.0

